project/src/main/java/com/google/sps/data/MapsLibrary.java

```diff
@@ -1,27 +1,60 @@
 package com.google.sps.data;
 
-/* 
-Basic HTML:
-    <script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY"></script>
-    <body onload="createMap();">
-    <div id="map"></div>
-*/
-
-/*
-Basic CSS:
-    #map {
-        border: thin solid black;
-        height: 500px;
-        width: 500px;
-    }
-*/
-
-/*
-Basic Javascript:
-    function createMap() {
-        const map = new google.maps.Map(
-        document.getElementById('map'),
-        {center: {lat: 37.422, lng: -122.084}, zoom: 16});
+import java.util.stream.Collectors;  
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.sps.data.ProductWithAddress;
+
+import java.util.*; 
+import java.lang.*; 
+
+public class MapsLibrary {
+    private static Double oLat;
+    private static Double oLng;
+    private static Double dLat;
+    private static Double dLng;
+
+    public static Double distance(GeoPt origin, GeoPt destination) {
+        //Convert coordinates to radians instead of degrees
+        oLat = Math.toRadians(Double.parseDouble(new Float(origin.getLatitude()).toString()));
+        oLng = Math.toRadians(Double.parseDouble(new Float(origin.getLongitude()).toString()));
+        dLat = Math.toRadians(Double.parseDouble(new Float(destination.getLatitude()).toString()));
+        dLng = Math.toRadians(Double.parseDouble(new Float(destination.getLongitude()).toString()));
```

Comment:
this works but can be done a little cleaner. instead of casting to string and then converting to double, the Float class has a member called doubleValue().  so:

`new Float(destination.getLatitude()).doubleValue()
`
should be equivalent without the string cast and parsing